resources/aws/ec2/vm.go

```diff
@@ -44,7 +52,9 @@ func NewInstance(e aws.Environment, name string, args InstanceArgs, opts ...pulu
 			"Name": e.Namer.DisplayName(255, pulumi.String(name)),
 		},
 		InstanceInitiatedShutdownBehavior: pulumi.String(e.DefaultShutdownBehavior()),
+		MetadataOptions:                   ec2.InstanceMetadataOptionsPtr(metadataOptions),
```

Comment:
Does this line behave as expected when metadataOptions is `nil`?

Reply:
if the arg is nil then the function return nil

Reply:
By default `metadataOptions` is `nil`